File: go/eliuds-eggs/eliuds_eggs.go
// exercise: eliuds eggs - count binary digits.
package eliudseggs

// Count the bits in a number
// @param displayValue binary representation of the eggs in their locations converted to decimal
// @returns: count of 1's in the binary.
func EggCount(displayValue int) int {
	var count int = 0
	var currentValue int = displayValue

	for currentValue != 0 {
		if currentValue&0x01 == 1 {
			count++
		}
		currentValue = currentValue >> 1
	}
	return count
}

File: php/eliuds-eggs/EliudsEggs.php
<?php
// Eliuds Eggs Example.

declare(strict_types=1);

// Egg counter class
class EliudsEggs
{
    // Get the bits from a number and count the 1s
    // @param $displayValue: Decimal value of the bits
    // @returns: Number of 1's in the number.
    public function eggCount(int $displayValue): int
    {
        $value = $displayValue;
        $count = 0;
        while ($value != 0) {
            if ($value & 0b00000001 > 0) {
                $count += 1;
            }
            $value = $value >> 1;
        }
        return $count;
    }
}
